### PR TITLE
Improving Transform: Part 1 (src - line 1 to 500)

### PR DESCRIPTION
**Pull request by saeeedtorabi**
- improved src code formatting
- improved documentation
- added new tests
- fixed few minor bugs

### CODE REVIEW

File: core/geometry/src/geometry3d/Transform.ts [hunk 1]
```diff
@@ -224,183 +255,247 @@ export class Transform implements BeJSONFunctions {
...
+      result
+    );
   }
-  /** Inverse transform the input homogeneous point.
-   * * Return as a new point or in the optional result.
-   * * returns undefined if the matrix part if this Transform is singular.
+  /**
+   * If for a point `p` we have `Tp = M*p + w*o = weightedPoint` (where `weightedPoint` is the transformed point), then
+   * `p = MInverse * (weightedPoint - w*o)`. This function returns a Point4d where first 3 elements are the original
+   *  point `p` if `weightedPoint` is the transformed point (`Tp = weightedPoint`) and the last element is `w`.
```
Comment (dassaf4) [inline core/geometry/src/geometry3d/Transform.ts:470]:
```suggestion
   * Multiply the homogenous point by the inverse transform.
   * * If for a point `p` we have `T*p = M*p + o = q`, then `p = MInverse*(q - o)` so `TInverse` has matrix part `MInverse` and origin part `-MInverse*o`.
   * * This method computes `TInverse*p = MInverse*p - w*MInverse*o` and returns the `Point4d` formed by `TInverse*p` in the first three coordinates, and `w` in the fourth.
   * * Logically, this is multiplication by the inverse of the 4x4 matrix formed from the 3x4 instance augmented with fourth row 0001. This is equivalent to the 4x4 matrix formed in similar fashion from the inverse of this instance.
```

File: core/geometry/src/geometry3d/Transform.ts [hunk 2]
```diff
@@ -224,183 +255,247 @@ export class Transform implements BeJSONFunctions {
...
+      w,
+      result
+    );
   }
-
-  /** Return product of the transform's inverse times a point (point given as x,y,z) */
+  /**
+   * If for a point `p = (x,y,z)` we have `Tp = M*p + o = point` (where `point` is the transformed point), then
+   * `p = MInverse * (point - o)`. This function returns the original point `p` if `point` is the transformed
+   * point (`Tp = point`).
```
Comment (dassaf4) [inline core/geometry/src/geometry3d/Transform.ts:487]:
```suggestion
   * Multiply the point by the inverse transform.
   * * If for a point `p` we have `T*p = M*p + o = q`, then `p = MInverse*(q - o)` so `TInverse` has matrix part `MInverse` and origin part `-MInverse*o`.
```

**Review by dassaf4 — APPROVED** [review]
Looks good. Made several suggestions to improve comment consistency.

File: core/geometry/src/geometry3d/Transform.ts [hunk 3]
```diff
@@ -138,20 +157,20 @@ export class Transform implements BeJSONFunctions {
...
+      this._matrix.clone()
+    );
   }
-  /** Return a copy of this Transform, modified so that its axes are rigid
-   * * The first axis direction named in axisOrder is preserved
-   * * The plane of the first and second directions is preserved, and its vector in the rigid matrix has positive dot product with the corresponding vector if the instance
-   * * The third named column is the cross product of the first and second.
+  /**
+   * Return a modified copy of `this` Transform so that its `matrix` part is rigid (`origin` part is untouched).
+   * * For details of how the matrix is modified to rigid, see documentation of `Matrix3d.axisOrderCrossProductsInPlace`
```
[the remark this answers is not in the document]
Comment (saeeedtorabi) [reply core/geometry/src/geometry3d/Transform.ts:165]: Looks like I need @see [[method_name]] or if the method is in another class @see [[class_name.method_name]]

**Pull request merged**